### PR TITLE
ENG-18635: do not discard segments when closing last cursor

### PR DESCRIPTION
**Pull request by rdykiel**

### CODE REVIEW

File: src/frontend/org/voltdb/utils/PersistentBinaryDeque.java [hunk 1]
```diff
@@ -1006,6 +1006,12 @@ public synchronized void closeCursor(String cursorId) {
         }
         reader.close();
 
+        // If we closed the last cursor, leave the PBD segments as they are, which
```
Comment (manjujames) [inline src/frontend/org/voltdb/utils/PersistentBinaryDeque.java]: The comment below seems to indicate that this is required for some weird DR use case. Will this change break DR?

Comment (rdykiel) [reply src/frontend/org/voltdb/utils/PersistentBinaryDeque.java]: I have no idea...

Comment (manjujames) [reply src/frontend/org/voltdb/utils/PersistentBinaryDeque.java]: I talked to Walter and looked at this again. This is probably required by DR. So may be add a flag to this call or something like that to delete segments or not based on the flag.

Comment (manjujames) [reply src/frontend/org/voltdb/utils/PersistentBinaryDeque.java]: Actually, a better option might be to expose close in the API and make that close cleanup readers map as well. Then PBD.closeCursor can be used just for this DR special case. May be even change the method name to make it clear that it does more than just closing the cursor.

**Pull request merged**